conda/core/package_cache_data.py

```diff
@@ -816,40 +824,48 @@ def execute(self):
                 )
                 futures.append(future)
 
-            for completed_future in as_completed(futures):
+            while futures:
                 try:
-                    prec_or_spec = completed_future.result()
-                except BaseException as e:
-                    # Special handling for exceptions thrown inside future, not
-                    # by futures handling code. Cancel any download that has not
-                    # been started. In-progress futures will continue.
-                    # Could use executor.shutdown(cancel_futures=True) instead?
-                    for future in futures:
-                        future.cancel()
-
-                    # faster handling of KeyboardInterrupt e.g.
-                    if not isinstance(e, Exception):
-                        raise
-
-                    # break, not raise. CondaMultiError() raised if exceptions.
-                    break
-
-                cache_action, extract_action = self.paired_actions[prec_or_spec]
-                extract_future = extract_executor.submit(
-                    do_extract_action,
-                    prec_or_spec,
-                    extract_action,
-                    progress_bars[prec_or_spec],
-                )
-                extract_future.add_done_callback(
-                    partial(
-                        done_callback,
-                        actions=(cache_action, extract_action),
-                        exceptions=exceptions,
-                        progress_bar=progress_bars[prec_or_spec],
-                        finish=True,
-                    )
-                )
+                    for completed_future in as_completed(futures, timeout=1):
```

Comment:
According to one stackoverflow post, at least subprocess waits work better when there is a timeout. Perhaps this is also true for as_completed?